src/requestforattestation/RequestForAttestation.ts

```diff
@@ -107,22 +113,55 @@ export default class RequestForAttestation implements IRequestForAttestation {
    */
   public constructor(
     claim: IClaim,
-    legitimations: AttestedClaim[],
-    identity: Identity,
+    legitimations: IAttestedClaim[],
+    identity?: Identity,
+    claimOwner?: NonceHash,
+    claimerSignature?: string,
+    claimHashTree?: object,
+    ctypeHash?: NonceHash,
+    rootHash?: Hash,
     delegationId?: IDelegationBaseNode['id']
   ) {
-    if (claim.owner !== identity.address) {
-      throw Error('Claim owner is not identity')
-    }
-    this.claim = claim
-    this.claimOwner = generateHash(this.claim.owner)
-    this.ctypeHash = generateHash(this.claim.cType)
-    this.legitimations = legitimations
-    this.delegationId = delegationId
+    if (identity !== undefined) {
+      if (claim.owner !== identity.address) {
+        throw Error('Claim owner is not identity')
+      }
+      this.claim = claim
+      this.claimOwner = generateHash(this.claim.owner)
+      this.ctypeHash = generateHash(this.claim.cTypeHash)
+      this.legitimations = legitimations.map((legitimation: IAttestedClaim) =>
```

Comment:
I don't think these need to be deep cloned. I also feel the `fromObject` method is a better place for this.

Reply:
outsource to from...() methods